packages/di/src/domain/DIContext.ts

```diff
@@ -92,6 +93,13 @@ export class DIContext extends Map<any, any> implements ContextMethods {
   async emit(eventName: string, ...args: any[]) {
     return this.injector?.emit(eventName, ...args);
   }
+
+  runInContext(next: Function) {
+    return runInContext(this, () => {
+      next = this.injector?.alter("$alterRunInContext", next, this);
```

Comment:
I suggest to provide ability to use `async` `$alterRunInContext` hook. It might be useful in some cases.

```suggestion
    return runInContext(this, async () => {
      next = await this.injector?.alterAsync("$alterRunInContext", next, this);
```